Map Loader/Program.fs

```diff
@@ -185,61 +234,99 @@ let rec start () =
+      let cmd_parts = command.Split(' ')
+      colorprintfn "Using special command $yellow[%s]" command
+      match command.ToLower() with
+      | "@date desc" ->
+        maps_files
+        |> List.sortByDescending(fun f -> f.CreationTime)
+        |> List.map(fun f -> Path.GetFileNameWithoutExtension(f.Name))
+        |> displayMaps Console.CursorTop 0
+      | "@date asc" -> 
+        maps_files
+        |> List.sortBy(fun f -> f.CreationTime)
+        |> List.map(fun f -> Path.GetFileNameWithoutExtension(f.Name))
+        |> displayMaps Console.CursorTop 0
+      | _ ->
+        cmd_parts.[0]
+        |> colorprintfn "$red[Unrecognised special command %s]"
+        start ()
+  else
+    let filter = parseCommand(command)
+    
+    let filteredMaps = maps |> List.filter(fun x -> filter.IsMatch x)
     if filteredMaps.Length = 0 then
       colorprintfn "$red[%A doesn't match any maps]" filter
       start ()
     else
       colorprintfn "There are %i results" filteredMaps.Length
-      displayMaps filteredMaps Console.CursorTop 0
-  with _ ->
-    colorprintfn "$red[Regex was invalid.]"
-    start()
+      displayMaps Console.CursorTop 0 filteredMaps
+  //with _ ->
+  //  colorprintfn "$red[Regex was invalid.]"
+  //  start()
 
 let loading = [|"/"; "-"; "\\"; "|"|]
 
-let unixtime = (DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalSeconds |> int64) - 604800L
 let mergeMaps = Map.fold (fun acc key value -> Map.add key value acc)
 
 let initialiseCache () =
   let wc = new WebClient()
   let mutable alreadyParsedUsers:Map<int,userinfo> = Map.empty
-  let _m = maps |> Array.filter(fun x -> cache.ContainsKey x |> not)
-  colorprintfn "$cyan[Updating cache]"
-  colorprintf "$gray[Updating users]"
-  Console.CursorTop <- Console.CursorTop - 1
-  Console.CursorLeft <- 0
-  _m |> Array.iteri(fun i map ->
-    colorprintf "$cyan[Updating cache (%i/%i) %s %s                  ]" (i+1) _m.Length loading.[i%4] map
-    Console.CursorLeft <- 0
+  let _m = maps |> List.filter(fun x -> cache.ContainsKey x |> not)
+  //colorprintfn "$cyan[Updating cache]"
+  //colorprintf "$gray[Updating users]"
+  //Console.CursorTop <- Console.CursorTop - 1
+  //Console.CursorLeft <- 0
+  _m |> List.iteri(fun i map ->
+    Console.Title <- sprintf "Map Loader %s [initialising cache %i/%i]" version (i+1) (_m.Length)
+    //colorprintf "$cyan[Updating cache (%i/%i) %s %s                  ]" (i+1) _m.Length loading.[i%4] map
+    //Console.CursorLeft <- 0
     try
       let res = wc.DownloadString(sprintf "https://api.skylarkx.uk/mapinfo?map=%s" map)
       let api = res |> JsonConvert.DeserializeObject<apimapinfo>
       cache <- cache.Add(map, {userid=api.forum_user_id; forumid=api.forum_id})
-      //users <- users.Add(map, {username=api.forum_user.username; userid= api.forum_user.id; display=api.forum_user.display; updated=api.forum_user.updated})
+      //printfn "'%s':%A %b" (api.map) (api.forum_user) (api.map = null)
+
+      if api.map <> null then
```

Comment:
use `isNull api.map` 
see also: https://latkin.org/blog/2015/05/18/null-checking-considerations-in-f-its-harder-than-you-think/